### DISCUSSION TITLE
Bug when matching brackets

### DISCUSSION THREAD

**Discussion by margual56** [pest-parser/pest#992] [Q&A] [answered]
## Describe the bug
I can't seem to match either '{' or '[' using the parser integrated in [the pest website](https://pest.rs/?g=N4Ig5gTghgtjURALhAdQBIEkAqBRAygAoCCAwrgAQC8FA%2BsBQDohMsA%2BrjjALsxRwDlcqADKYhFAL6MAdrIAOEAPbyAphG4BLVQGdqFAAINmAbT4A-CsQEBNAFQVLzALp9pMkABoQmmfICu3MggJtgAFpp6kRRQFDJK3KrOIJJAA#editor). 

## To Reproduce
Click [here](https://pest.rs/?g=N4Ig5gTghgtjURALhAdQBIEkAqBRAygAoCCAwrgAQC8FA%2BsBQDohMsA%2BrjjALsxRwDlcqADKYhFAL6MAdrIAOEAPbyAphG4BLVQGdqFAAINmAbT4A-CsQEBNAFQVLzALp9pMkABoQmmfICu3MggJtgAFpp6kRRQFDJK3KrOIJJAA#editor) to try it out.

### Grammar
```pest
WHITESPACE = _{ " " | "\t" | NEWLINE }

properties = { "[" ~ ASCII* ~ "]" }
```

### Text to match
`[This is a note]`

### Output
```
 --> 1:1
  |
1 | [This is a note]
  | ^---
  |
  = expected properties
```

### Screenshot
![image](https://github.com/pest-parser/pest/assets/30444886/6111d8d7-3cef-470d-a93a-594889d65d0b)

### Extra
I've also tried with: `properties = { "{" ~ ASCII* ~ "}" }` and `properties = @{ "[" ~ ASCII* ~ "]" }`, but it gives the same error.

## Expected behavior
Clearly, it should be able to match the given string... right?

**Comment by tomtau** [comment, marked as answer]
PEGs are eager: https://pest.rs/book/grammars/peg.html#eagerness
so `ASCII*` will also match `]`, and the whole expression will fail to match.

You can try `ASCII_ALPHANUMERIC` (https://pest.rs/book/grammars/built-ins.html#ascii-rules) instead:
```
properties = { "[" ~ ASCII_ALPHANUMERIC* ~ "]" }
```

or you can define your own rule of allowed characters that can be inside brackets

**Reply by margual56** [reply]
Thank you for the answer!

**Reply by margual56** [reply]
![image](https://github.com/pest-parser/pest/assets/30444886/894765ef-e04d-4dd7-8c90-3e7b28b1af3f)

Sadly it still does not work :(

**Reply by tomtau** [reply]
Do you have the WHITESPACE rule defined?

**Reply by tomtau** [reply]
https://pest.rs/?g=N4Ig5gTghgtjURALhAdQBIEkAqBRAygAoCCAwrgAQC8FA%2BsBQDohMsA%2BrjjALsxRwDlcqADKYhFAL6MAdrIAOEAPbyAphG4BLVQGdqFBswDafAH4Vi%2BUpky1iIwumICAqgFlcAJUykAVBXNmAF0%2BaRkQABoQTRl5AFduZBAjbAALTT0MiigKGSVuVSCQSSA#editor

**Reply by margual56** [reply]
Nope, I took it out for the demo here, that was the issue 😆 

Thank you so much!
